Aztec/Classes/NSAttributedString/Conversions/HTMLConverter.swift

```diff
@@ -78,3 +93,21 @@ public class HTMLConverter {
     }
     
 }
+
+// MARK: - Helpers
+
+private extension HTMLConverter {
+    func hasBodyNode(_ nodes: [Node]) -> Bool {
+        if nodes.isEmpty {
+            return false
+        }
+
+        switch nodes.first {
+        case let element as ElementNode where element.name == Element.body.rawValue:
+            return true
+        default:
+            let remainingChildren = Array(nodes.dropFirst())
```

Comment:
I'd suggest to change this method slightly to avoid re-creating the array for each element dropped.

Array slices are really easy to use, and the change here would fit their purpose imho (since `dropFirst()` creates a slice).

From https://developer.apple.com/documentation/swift/arrayslice:

>The ArraySlice type makes it fast and efficient for you to perform operations on sections of a larger array. Instead of copying over the elements of a slice to new storage, an ArraySlice instance presents a view onto the storage of a larger array. And because ArraySlice presents the same interface as Array, you can generally perform the same operations on a slice as you could on the original array.

You can reformulate the code like so:

```swift
    func hasBodyNode(_ nodes: [Node]) -> Bool {
        return hasBodyNode(ArraySlice<Node>(nodes))
    }
    
    private func hasBodyNode(_ nodes: ArraySlice<Node>) -> Bool {
        if nodes.isEmpty {
            return false
        }
        
        switch nodes.first {
        case let element as ElementNode where element.name == Element.body.rawValue:
            return true
        default:
            return hasBodyNode(nodes.dropFirst())
        }
    }
```

Reply:
That's a great improvement! Thank you!